### DISCUSSION TITLE
请问如何写hosts

### DISCUSSION THREAD

**Discussion by Kainichy** [XIU2/CloudflareSpeedTest#357] [讨论]
具体地，hosts里填写的域名应该是什么呢？
因为Cloudflare给出的DNS record都是A，并非CNAME……我不知道该如何操作hosts以解析最快的IP地址。

**Comment by XIU2** [comment]
写什么域名取决于你的需求啊，你常访问的哪个国外网站用的是 Cloudflare CDN，那你就写这个网站的域名。

**Comment by Kainichy** [comment]
懂了，但是如果这样我倒觉得没必要。因为我的愿景就是所有用Cloudflare CDN的网站都上最快IP地址……

> 写什么域名取决于你的需求啊，你常访问的哪个国外网站用的是 Cloudflare CDN，那你就写这个网站的域名。

**Comment by XIU2** [comment]
你这要求确实无法实现，因为无法获得所有 Cloudflare CDN 的网站域名，我这个项目只是我个人的小需求，自给自足。

**Comment by XIU2** [comment]
另外对一些使用 Cloudflare CDN 的站长也有用。

**Comment by Kainichy** [comment]
我自己就是哈哈。
btw，您使用的测速png是个什么样的文件呢？我尝试打开但是文件损坏了：
![image](https://user-images.githubusercontent.com/44992049/99139327-9ef64900-2672-11eb-918f-3c8fbf5e3df6.png)

**Comment by XIU2** [comment]
@SkyYkb 就只是在 Linux 上生成的 300MB 空文件，只是用来下载测速的。

**Comment by Kainichy** [comment]
这，行吧……
祝好，再见👋。

**Comment by XIU2** [comment]
我也只是把我经常访问的网站域名（Cloudflare CDN）加到 Hosts 里，自己维护一份自用的。
一劳永逸是没办法了。

**Comment by XIU2** [comment]
对了，根据反馈，我也发现有一些人使用 Cloudflare CDN 来科学上网（用我的工具测速最快的 CDN IP，来提高代理速度）。

> 通过 Cloudflare Workers 服务创建代理网站。。。
> 代理服务器用网站伪装的同时，还可以搭配 Cloudflare CDN 来解决被墙 IP。。。

**Comment by YujiaCheng1996** [comment]
> 懂了，但是如果这样我倒觉得没必要。因为我的愿景就是所有用Cloudflare CDN的网站都上最快IP地址……
> 
> > 写什么域名取决于你的需求啊，你常访问的哪个国外网站用的是 Cloudflare CDN，那你就写这个网站的域名。

单个ip可能占不满本地带宽，也可能会拥堵，单点故障等。
还有可能因为长时间大流量访问同一个境外ip被运营商特别关照和谐。。。
当然过多ip也没必要。

**Comment by XIU2** [comment]
@SkyYkb  你的愿望实现了，一劳永逸的方法 #71
